sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1577,11 +1577,16 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
         .. versionadded:: 0.23
 
     scoring : str or callable or None, default='loss'
-        Scoring parameter to use for early stopping. It can be a single
-        string (see :ref:`scoring_parameter`) or a callable (see
-        :ref:`scoring`). If None, the estimator's default scorer is used. If
-        ``scoring='loss'``, early stopping is checked w.r.t the loss value.
-        Only used if early stopping is performed.
+        Scoring parameter to use for early stopping. Only used if early stopping
+        is performed. Options:
+
+        - str: see :ref:`scoring_string_names` for options.
+        - callable: a scorer callable object (e.g., function) with signature
+          ``scorer(estimator, X, y)``. See :ref:`scoring_callable` for details.
+        - `None`: the :ref:`coefficient of determination <r2_score>`
+          (:math:`R^2`) is used.
+        - 'loss': early stopping is checked w.r.t the loss value.
```

Comment:
Hmm maybe I don't understand this parameter. I thought this was the metric to use to see if we should stop early, it's not a score to return..? In which case 'compute score' seems misleading?

Reply:
> I am more confused about this than before.

The meat of the early stopping code seems to be here:

https://github.com/scikit-learn/scikit-learn/blob/ebfda90cd7bdf89e7222510f8b92fbe716d7f9d5/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py#L730-L756

AFAICT, we want to check if there has been improvement in the score of the predictions, and if not we stop early (before we reach max leaves or something). For 'loss' we just use the loss, otherwise a scorer function is used. Both calculate on predictions, but scorers have the signature `(est, X, y)`, so we do it on a subset of train to avoid calling predict constantly.

Reply:
Yes, after looking at it again I would agree. 

The `scoring` param here is used differently than in the CV models or in cross validation, which is only for determining how we calculate early stopping (if it is enabled). It's the parameter name, that is confusing. But this cannot be changed so easily, so we need to live with it.

Going back to the documentation I see that it doesn't promise anything more than the scoring is used in early stopping: `Scoring parameter to use for early stopping.` Still, I wonder if this misunderstanding could be prevented for other people by rewording into `Scoring to use for calculating early stopping.` Or alternatively add as a last sentence, that this `scoring` param differs from the way scoring is defined in the Glossary. But it's also good as it is.

Reply:
I see your point. I've altered it slightly to "Scoring method to use ..." because the 'parameter' part seemed redundant and possibly confusing.

I think it is clear enough in the parameter docstring that this is used only for early stopping. What we could do though is add an extra sentence in the glossary about how `scoring` is sometimes used to specify scoring method to use for early stopping as well?

Reply:
> What we could do though is add an extra sentence in the glossary about how scoring is sometimes used to specify scoring method to use for early stopping as well?

Oh yes, that's good. :+1: